### PR TITLE
The monkey portrait decoration is now a set of random pictures that also act as persistent photo frames. Every station map gets one.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
So, I was getting very very very bored while working on some other PRs that I suddenly thought about that one monkey portrait object with a funny description found by the metastation bar for no apparent reason and then my thoughts drifted toward the Treachery of Images and then back to ss13 picture frames. "Do picture frames have persistence support?". Turns out they do. So I decided to make that portrait more than just a static deco object (it has 4 variants now, chosen randomly on init) and also able to store persistent photos for the amusement of future bartenders.

## Why It's Good For The Game
Makes the portrait a special kind of decoration. Small, but unique in a way. I also tested it over multiple local tests and reboots. Smooth like butter.

## Changelog

:cl:
expansion: The Mr. Deempisi Portrait plaque has been replaced with a very special picture frame: A small picture of one out four random things (Mr. Deempisi included) that can have a photo affixed over it. The photo will persist between rounds on that specific station until removed of if the frame is destroyed. Each station gets one, all but one (Kilo is too damn cramped) located in the bar behind the bartender counter. It's also more artsy than other picture frames now.
spriteadd: Desouled the picture frame sprites.
/:cl:

### CODE REVIEW

**Comment by Zytolg** [comment]
Some of your mapdiffs are pretty big given the stated scope of this change. It appears that you're either unintentionally removing or adding some objects from the map, or you're removing or adding things from the maps without documentation. Either way, you'll need to check out what's going on in Deltastation, Iceboxstation (z2), and Kilostation. I wasn't able to spot what else got changed on Kilostation and Deltastation, but given how those diffs rendered, Delta's additions/removals appear to be south of the station, and Kilo's appears to have stuff throughout (the diff is about the majority of the station). You might need to check walls. Icebox might also have these issues, but at least I was able to spot some things here. 

Mapthing

You appear to be removing an air alarm here, as well as some AI Displays. As none of these are documented changes, nor do they appear to be being replaced with the new portraits, I have to ask why these changes are needed.

**Comment by Mickyan** [comment]
Pretty clever grabbing the map name for the persistence id, wondering if it can be made more generic so it can be used for other frames without copy pasta

**Comment by Ghommie** [comment]
@Zytolg I had to (re)move a few things like posters around to make space for the portrait. I hope this isn't an issue.

**Comment by Ghommie** [comment]
> 
> 
> Pretty clever grabbing the map name for the persistence id, wondering if it can be made more generic so it can be used for other frames without copy pasta

It could be a macro, but so far only the bar portrait frame uses it.

**Comment by Zytolg** [comment]
> @Zytolg I had to (re)move a few things like posters around to make space for the portrait. I hope this isn't an issue.

Not at all, this is a case of map diff bot being quirky. In the future though, it might help to include pictures of the changed areas so that it's easy to identify at a glance.

File: code/controllers/subsystem/persistence.dm [hunk 1]
```diff
@@ -262,6 +262,19 @@ SUBSYSTEM_DEF(persistence)
...
+	var/frame_path = file("data/photo_frames.json")
+	if(!fexists(frame_path))
+		return
+
+	var/frame_json = json_decode(file2text(frame_path))
+	fdel(frame_path)
+	frame_json -= identifier
+
+	frame_json = json_encode(frame_json)
+	WRITE_FILE(frame_path, frame_json)
```
Comment (Watermelon914) [inline code/controllers/subsystem/persistence.dm:276]: only fdel right before you write to the file to make sure you don't lose the data if something runtimes inbetween

**Review by Maurukas — APPROVED** [review]
Mapdiff bot doesn't like it, but everything looks fine. From a mapping perspective this is good to go.

**Review by Watermelon914 — APPROVED** [review]
needs spritetainer review now

**Pull request merged**